lib/sycamore/sycamore/executor.py

```diff
@@ -87,6 +89,23 @@ def execute_iter(self, plan: Node, **kwargs) -> Iterable[Document]:
 
         plan.traverse(visit=lambda n: n.finalize())
 
+    def _prepare(self, plan: Node):
+        from queue import Queue
+
+        pending: Queue[Callable] = Queue()
```

Comment:
```
# Some prepare operations need to be executed in phases, so here's a queue for that
```

Reply:
Done, made it longer to cover the discussion and example.